### DISCUSSION TITLE
Sehr viel seltener Werte vom HM-600 nach Update

### DISCUSSION THREAD

**Discussion by mar1ged** [tbnobody/OpenDTU#375] [Q&A] [answered]
Ich hatte "lange" die bad9031 eingesetzt und war damit sehr glücklich. Jetzt habe ich das Update auf 854af46 bzw. 8ba0a65 vorgenommen und erhalte deutlich seltener Werte über die DTU. Den Großteil des Tages sehe ich ein ein "Data Age" von mehreren 1000 Sekunden, teilweise wird ein Wert ausgelesen und in der Oberfläche angezeigt. Dann ist aber wieder sehr lange Ruhe und es kommen keine neuen Werte. Obwohl definitiv genügend Sonne auf die zwei Panele trifft.

Den Status des Hoymiles habe ich überprüft (Leiter rauf und runter auf die Garage ;-)), er funktioniert einwandfrei und blinkt auch im erwarteten Rhythmus grün.  

Sind ggf. Probleme mit der neueren Version bekannt ?

Und eine generelle Wissenfrage noch: unterstützt der Hoymiles _zwei_  OpenDTU Instanzen ? Ich würde dann einfach vor einem Update eine zweite Hardware mit dem "Kandidaten" installieren, beide parallel betreiben und schauen, ob die neue Version auch für mich gut funktioniert. Erst nach einer Testphase würde ich dann meine primäre Hardware aktualisieren.

**Comment by Solarteur** [comment]
Seit knapp drei Tagen läuft bei mir die Version "854af46" - auch mit einem HM-600: Nach wie vor werden alle dreißig Sekunden die Werte aktualisiert.

**Comment by tbnobody** [comment]
Zwischen den von dir genannten Versionen wurde eigentlich nichts an der Auslese Routine geändert. 

Der Inverter unterstützt den parallelen Betrieb von 2 DTU's NICHT.

**Reply by mar1ged** [reply]
Danke @tbnobody und @Solarteur. Habt ihr vielleicht noch einen Tipp was ich prüfen kann ?

Ich würde sonst morgen Mal die Zweithardware in Betrieb nehmen und die primäre vom Strom lassen

**Reply by Solarteur** [reply]
Ein Tausch der Hardware ist sicher eine gute Idee. Wie ist deine Hardware aufgebaut: Platine und verlötet oder gesteckte Kabel? Schwachstelle kann auch das nRF- Modul sein.

**Reply by mar1ged** [reply]
Da ich noch in der Test- und Lernphase bin, nutze ich noch die Jumperkabel zwischen ESP32 und nRF-Modul. An sich zeigt die Weboberfläche aber eine Verbindung zum nRF-Modul an, wobei das aber natürlich nichts über die Qualität der Verbindung oder eventuelle Schäden aussagt.
![Screenshot_2022-11-22-22-58-18-747-edit_com android chrome](https://user-images.githubusercontent.com/16045549/203429879-cd1f6a68-8873-4b65-811c-c9f0992d084f.jpg)

**Reply by mar1ged** [reply]
Zwischenstand: leider auch mit der anderen ESP32 Hardware kein Kontakt, die ist auf 854af46. Sollte seit gut einer Stunde schon online sein, aber ist es leider nicht.

Da werde ich mir nachher wohl mal die Leiter holen ...

**Comment by mar1ged** [comment]
Noch eine Verständnisfrage. Ich habe ein Downgrade auf bad9032 durchgeführt. Die funktionierte zuvor, leider habe ich aber auch hier keine Verbindung. 
Wenn ich den Screenshot ansehe, *könnte* das bedeuten, dass zumindest eine Verbindung aufgebaut werden konnte.
Zumindest kann ich mich daran erinnern, dass ich im Log schon einen Hinweis der Art "noch nix da, weil bisher keine Verbindung" gesehen habe. Oder ist das ein Text, der in der Version gar nicht erscheinen kann ?

![Screenshot_2022-11-23-09-34-13-844-edit_com android chrome](https://user-images.githubusercontent.com/16045549/203504058-5de45bd8-21e8-4f1e-b712-ee299c719d85.jpg)

**Reply by mar1ged** [reply]
Jetzt verstehe ich die Welt nicht mehr...
Folgendes habe ich getan:
- Status des HM-600 geprüft, er blinkt artig im kurzen Abstand grün
- Hardware auf "last known good" bad9031 geflashed
- Werkseinstellungen
- über OpenDtu WLAN verbunden
- Zeit synchronisiert
- Inverter konfiguriert
- Abstand auf ca. 2m zwischen DTU und Inverter reduziert 

Innerhalb ca. 30 Minuten wurde eine Verbindung aufgebaut. Dadurch konnte ich sehen, dass über den Tag eifrig produziert wurde. 
Da dann ein paar Minuten keine Aktualisierung mehr möglich war, habe ich den PA Level erhöht.
Seitdem leider keinerlei Updates mehr, auch nachdem ich wieder reduziert habe und die Sonne fleißig schien

![Screenshot_2022-11-23-15-51-48-811-edit_com android chrome](https://user-images.githubusercontent.com/16045549/203588267-0dbdb80b-bb08-4958-a41f-8c9e97ff20df.jpg)

**Comment by tbnobody** [comment, marked as answer]
Das klingt irgendwie nach schlechtem Empfang. Nur so zum Vergleich. Wenn ich den ESP neu starte dauert es ca. 10-15 Sekunden bis die Zeit synchronisiert ist. Das Senden der Anfragepakete findet dann sofort statt. Direkt auf diese Pakete bekomme ich die entsprechenden Antworten. Also nach einem Reboot ca. 60 Sek. bis alle Daten (Statistik + Eventlog + Device Info) vorhanden ist. Was siehst du denn in der seriellen Konsole? Dort werden noch mehr Debug Meldungen ausgegeben.

**Reply by mar1ged** [reply]
Ja, das könnte für sein..Verstehe nur die Ursache nicht, die Entfernung ist kurz und ich habe ja bewusst die Variante mit voller Antenne und nicht nur einem kleinen Draht im Einsatz.

Werde mir das mit der Konsole anschauen, bisher wüsste ich nicht, wie ich das anstelle

**Reply by tbnobody** [reply]
> Werde mir das mit der Konsole anschauen, bisher wüsste ich nicht, wie ich das anstelle

Generell so:
https://github.com/tbnobody/OpenDTU/issues/332#issuecomment-1312483812

**Reply by mar1ged** [reply]
Danke für die Infos.

Mir ist aber auch eines aufgefallen: kann es sein dass manche nRF nicht so gut sind ? Ich habe das H-1-0781 und wenn ich bei PA über Low hinausgehe, kann ich wohl reproduzierbar die Verbindung killen

![Screenshot_2022-11-24-09-59-30-802-edit_com android chrome](https://user-images.githubusercontent.com/16045549/203747998-cd6bcfb2-9798-4578-9c89-9ab6cebd6d9b.jpg)

**Reply by tbnobody** [reply]
Jeb. Das wurde schon öfter angesprochen. Teilweise ist die Qualität auch nicht wirklich überragend. Von defekten Modulen über Lötbrücken usw. hab ich da schon alles gehört. Es scheint aber so, das Module von eByte eine bessere Verarbeitung und v.a. Schirmung haben. Sind aber etwas teurer.

**Reply by mar1ged** [reply]
Danke, das war mir entgangen bzw ich dachte ich hätte etwas gutes erwischt ;-)

Noch eine vielleicht dumme Frage zur seriellen Konsole: eigentlich sollte ich Debugging ja unter "idealen" Bedingungen machen, also zB mit einem guten Netzteil. Wenn ich jetzt den ESP per USB mit dem Rechner verbinde, büße ich das ja ein. Oder habe ich da einen Denkfehler ?

Wenn nicht, wäre für so einen Fall vielleicht eine syslog Anbindung etwas feines. Zwar bekommt man damit das "Booten" nicht komplett ins Log, aber den Rest
